node/pkg/fetcher/app.go

```diff
@@ -355,6 +360,11 @@ func (a *App) initialize(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+
+	// initialize channel for temporarily keeping local aggregates
+	localAggregatesChannel := make(chan LocalAggregate, LocalAggregatesChannelSize)
+	go a.localAggregatesChannelProcessor(ctx, localAggregatesChannel)
```

Comment:
how about moving this into fetcher.Run() function?

if using the ctx from fetcher.Run(), it'll be canceled accordingly when `app.StopFetcher` has been called

Reply:
good idea Nick. Just to confirm, I'm guessing you mean the Run function in `app.go`?

Reply:
yes, to achieve this
you might have to include the channel property into app structure

Reply:
how about just passing the channel to the `initialize()` function as an argument since the channel is used only in that function

Reply:
ah there is refresh message restarts the app. Then it makes sense to make it part of the app structure :)

Reply:
I couldn't get the point, can you explain in more detail?
`initialize` should be the function which initialize internal applications, so I personally think it's better to keep the parameters clean.

once internal applications are initialized, those apps start running once `startAll` has been called

my feedback was that execution and initialization should be separated

Reply:
sure, ping me after the update